packages/platform/src/HttpTest.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @since 0.56.3
+ */
+import type { Context } from "effect"
+import type * as Deferred from "effect/Deferred"
+import type * as Effect from "effect/Effect"
+import type * as Layer from "effect/Layer"
+import type * as Queue from "effect/Queue"
+import type * as Scope from "effect/Scope"
+import type * as HttpApp from "./HttpApp.js"
+import type * as HttpClient from "./HttpClient.js"
+import type * as HttpClientRequest from "./HttpClientRequest.js"
+import type * as HttpServer from "./HttpServer.js"
+import type * as HttpServerRequest from "./HttpServerRequest.js"
+import type * as HttpServerResponse from "./HttpServerResponse.js"
+import * as internal from "./internal/httpTest.js"
+
+/**
+ * @since 0.56.3
+ * @category models
+ */
+export interface HttpInMemoryServerQueue {
+  queue: Queue.Queue<
+    readonly [HttpClientRequest.HttpClientRequest, Deferred.Deferred<HttpServerResponse.HttpServerResponse>]
+  >
+}
+
+/**
+ * @since 0.56.3
+ * @category tags
+ */
+export const HttpInMemoryServerQueue: Context.Tag<HttpInMemoryServerQueue, HttpInMemoryServerQueue> =
+  internal.HttpInMemoryServerQueue
+
+/**
+ * A combinator accepting an `HttpApp` (therefore also a `HttpRouter`) and producing
+ * an effect the starts the server and returns an http client. The effect requires an
+ * `HttpClient` and `HttpServer` to be provided through the context. You'll probably want
+ * to use it in combination with `HttpTest.inMemoryLayer`, `NodeHttpTest.layer` or
+ * `BunHttpTest.layer`.
+ *
+ * @since 0.56.3
+ * @category combinators
+ */
+export const serve: <E, R>(app: HttpApp.Default<E, R>) => Effect.Effect<
+  HttpClient.HttpClient.Default,
+  E,
+  Exclude<R, HttpServerRequest.HttpServerRequest> | HttpClient.HttpClient.Default | HttpServer.HttpServer | Scope.Scope
+> = internal.serve
+
+/**
+ * Layer producing an `HttpClient` with prepended url of the running http server.
+ *
+ * @since 0.56.3
+ * @category layers
+ */
+export const clientLayer: Layer.Layer<
```

Comment:
Generally for layer apis, the naming starts with "layer" and then has a suffix.

If we move this to the HttpServer module, then this could be `layerTestClient`.

```suggestion
export const layerTestClient: Layer.Layer<
```